specs/protocol/configurability.md

```diff
@@ -0,0 +1,80 @@
+- **Admin Roles**: These roles can upgrade contracts, change role owners,
+  or update protocol parameters. These are typically cold/multisig wallets not
+  used directly in day to day operations.
+- **Service Roles**: These roles are used to manage the day-to-day
+  operations of the chain, and therefore are often hot wallets.
+
+## Consensus Parameters
+
+| Config Property                       | Description                                                                                                                  |
+|---------------------------------------|------------------------------------------------------------------------------------------------------------------------------|
+| Batch Inbox address                   | L1 address where calldata/blobs are posted (see [Batcher Transaction](../glossary.md#batcher-transaction)).                         |
+| Chain ID                              | Unique ID of Chain used for TX signature validation.                                                                         |
+| Challenge Period                      | Length of time for which an output root can be removed, and for which it is not considered finalized.                                                                                                                                                             |
+| Data Availability                     | Where L2 transaction calldata is posted.                                                                                     |
+| Fault Proof Window / Challenge Window | Duration network participants have to challenge the integrity of an output root.                                             |
+| Fee margin                            | Markup on transactions compared to the raw L1 data cost.                                                                     |
+| Proxy Fee vault contracts                   | Proxy contracts which point to implementation contracts that dictate how user fees are distributed.                                                                       |
+| Gas limit                             | L2 block gas limit.                                                                                                          |
+| Gas Token                             | Native token used to pay for gas.                                                                                            |
+| Genesis state                         | Initial state at chain genesis, including code and storage of predeploys (all L2 smart contracts). See [Predeploy](../glossary.md#l2-genesis-block).                                            |
+| L1 smart contracts                    | The chain’s L1 smart contracts.                                                                                              |
```

Comment:
I meant to comment on notion, but I think this entry is too broad/vague.
- I believe there are a set of contracts which are NOT changeable or configurable based on the version. These should not be in the configurability table. For example, changing the L1 bridge code should not be thought of as "configuring" the standard version, it should be thought of creating a new version.
- That above set could be most or all of the L1 contracts, I'm actually not sure if there's anything that does not fulfill the above. Unless @tynes or @protolambda can think of any important contracts which don't meet the above definition, I would propose just removing this entry from the table.

Reply:
I think we do still want to include L1 contracts here because there is a corresponding requirement that to be a standard chain you must be using a governance approved version of the L1 contracts.

In other words, the implementations are technically a configurable part of the OP Stack, but any change means you are not a standard chain, and we need to convey that requirement. If we remove this entry from the table we can't convey that.

Reply:
@mds1 I follow what you're saying, but I still feel pretty strongly that this is (or should be) a *versioning* difference, not a configuration difference.

I think the right way to convey the requirement is by requiring a version, which implies smart contracts, not by requiring certain bounds on config, which includes contracts.  As you said, "you must be using a governance approved _version_ of the L1 contracts".

It is a little less explicit in this PR's context, but I think it's also a valuable norm to say that changes to L1 contracts are a version difference, not a config difference.